src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java

```diff
@@ -129,4 +133,29 @@ public void updateInsolvencyLink(String contextId, String companyNumber,
         }
 
     }
+
+    private void updateSpecificFields(CompanyProfileDocument companyProfileDocument) {
+        Update update = new Update();
+        setUpdateIfNotNull(update, "data.etag",
+                companyProfileDocument.getCompanyProfile().getEtag());
+        setUpdateIfNotNull(update, "updated",
+                companyProfileDocument.getUpdated());
+        setUpdateIfNotNull(update, "data.links.insolvency",
+                companyProfileDocument.getCompanyProfile().getLinks().getInsolvency());
+        setUpdateIfNotNull(update, "data.links.charges",
+                companyProfileDocument.getCompanyProfile().getLinks().getCharges());
+        setUpdateIfNotNull(update, "data.has_insolvency_history",
+                companyProfileDocument.getCompanyProfile().getHasInsolvencyHistory());
+        setUpdateIfNotNull(update, "data.has_charges",
+                companyProfileDocument.getCompanyProfile().getHasCharges());
+        Query query = new Query(Criteria.where("_id").is(companyProfileDocument.getId()));
+        mongoTemplate.upsert(query, update, CompanyProfileDocument.class);
+    }
+
+    private Update setUpdateIfNotNull(Update update, String key, Object object) {
```

Comment:
Only updating the field if it's not null, otherwise won't be updated or modified